main.go

```diff
@@ -56,41 +55,31 @@ func setEnv() {
 	}
 
 	// configure metadata client
-	for {
-		time.Sleep(1000 * time.Millisecond)
-		m = metadata.NewClient(metadataUrl)
-		selfStack, err := m.GetSelfStack()
-		if err != nil {
-			logrus.Error("Error reading stack info: %v", err)
-			continue
-		}
-		EnvironmentName = selfStack.EnvironmentName
+	mClient, err := metadata.NewMetadataClient()
+	if err != nil {
+		logrus.Fatalf("Failed to configure rancher-metadata client: %v", err)
+	}
+	m = mClient
 
-		cattleUrl := os.Getenv("CATTLE_URL")
-		if len(cattleUrl) == 0 {
-			logrus.Error("CATTLE_URL is not set")
-			continue
-		}
+	cattleUrl := os.Getenv("CATTLE_URL")
+	if len(cattleUrl) == 0 {
+		logrus.Fatalf("CATTLE_URL is not set")
+	}
 
-		cattleApiKey := os.Getenv("CATTLE_ACCESS_KEY")
-		if len(cattleApiKey) == 0 {
-			logrus.Error("CATTLE_ACCESS_KEY is not set")
-			continue
-		}
+	cattleApiKey := os.Getenv("CATTLE_ACCESS_KEY")
+	if len(cattleApiKey) == 0 {
+		logrus.Fatalf("CATTLE_ACCESS_KEY is not set")
+	}
 
-		cattleSecretKey := os.Getenv("CATTLE_SECRET_KEY")
-		if len(cattleSecretKey) == 0 {
-			logrus.Error("CATTLE_SECRET_KEY is not set")
-			continue
-		}
+	cattleSecretKey := os.Getenv("CATTLE_SECRET_KEY")
+	if len(cattleSecretKey) == 0 {
```

Comment:
nit, and I'm not sure its idomatic or not, but you can do if cattleSecretKey == "" {} one less method call. Getenv returns an empty string if not found.